chapters/equations.tex

```diff
@@ -817,12 +817,18 @@ \section{Initialization, initial equation, and initial algorithm}\label{initiali
 variables) is equal.  Often, this is a nonlinear system of equations and therefore it may be necessary to provide appropriate guess values (i.e., \lstinline!start! values and
 \lstinline!fixed = false!) in order to compute a solution numerically.
 
-It may be difficult for a user to figure out how many initial equations have to be added, especially if the system has a higher index. A tool may add or remove initial equations
-automatically such that the resulting system is structurally nonsingular.  In these cases diagnostics are appropriate since the result is not unique and not necessarily what the user
-expects.  A missing initial value of a discrete-time variable which does not influence the simulation result, may be automatically set to the \lstinline!start! value or its default without
-informing the user.  For example, variables assigned in a \lstinline!when!-clause which are not accessed outside of the \lstinline!when!-clause and where \lstinline!pre! is not explicitly
-used on these variables, do not have an effect on the simulation.
+It may be difficult for a user to figure out how many initial equations have to be added, especially if the system has a higher index.
 \end{nonnormative}
+Based on these non-normative considerations we have the following rules:
+A tool may add or remove initial equations automatically such that the resulting system is structurally nonsingular.
+In these cases diagnostics are appropriate since the result is not unique and not necessarily what the user expects.
+\begin{itemize}
+\item A missing initial value of a discrete-time variable which does not influence the simulation result, may be automatically set to the \lstinline!start! value or its default without informing the user.
+For example, variables assigned in a \lstinline!when!-clause which are not accessed outside of the \lstinline!when!-clause and where \lstinline!pre! is not explicitly used on these variables, do not have an effect on the simulation.
```

Comment:
Looks good now.  Resolving.

Reply:
Wait, I don't see that we have this mismatch between _discrete-time expression_ and _discrete-time variable_.  That we don't have an index entry for _discrete-time variable_ is to me explained by the fact that the meaning follows directly from the meaning of _discrete-time expression_ and _variable_.  When I follow the added reference, I find what looks like a definition that doesn't exclude parameters and constants:
>   A \emph{discrete-time} variable \lstinline!vd! has a vanishing time derivative between events.
>  Note that this is not the same as saying that \lstinline!der(vd)=0! almost everywhere, as the derivative is not even defined at the events, and it is not legal to apply \lstinline!der! to discrete-time variables as they are not continuous. > During transient analysis the variable can only change its value at event instants (see \cref{events-and-synchronization}).

This feels like a bigger problem with our definitions, and I'm not sure the best solution is just to find another formulation to use here.

Reply:
But if you look at the three other bullet points: constant, parameter, and continuous-time variable you see that they are disjoint and don't include the less variable ones - in contrast to the corresponding points for expressions. That's why we can have continuous-time variables, but non-discrete-time-expressions. Similarly in 8.6 Initialization the discrete-time variables have rules that don't apply to parameters and constants.

So, vanishing time-derivative between events is either bad explanation since it is a necessary, but not sufficient, condition to be a discrete-time variable (technically a "continuous-time variable" may also satisfy it); or just a plain error.
The correct definition is sort of hinted and is Integer, Boolean (etc) with prefix discrete, or without prefix. And Real with prefix discrete or without prefix and assigned in a when-clause. Once it is defined we can then state discrete-time variables implies this property.

Reply:
Sounds like a plan!

Reply:
That is now a separate PR. So, is there any remaining issue with this PR?

Reply:
This is now a separate PR #2989 - so that we can close this one.

Reply:
Sure, we'd be able to close this once #2989 is fixed.

Reply:
Considering that we're homing in on merging #2989, let's consider this resolved.